x/auth/client/cli/sign.go

```diff
@@ -51,22 +58,37 @@ func makeSignCmd(cdc *amino.Codec, decoder auth.AccountDecoder) func(cmd *cobra.
 		}
 
 		if viper.GetBool(flagPrintSigs) {
-			printSignatures(stdTx)
+			if !printSignatures(stdTx) {
+				return fmt.Errorf("signatures validation failed")
+			}
 			return nil
 		}
 
 		name := viper.GetString(client.FlagName)
+		if name == "" {
+			return errors.New("required flag \"name\" has not been set")
+		}
 		cliCtx := context.NewCLIContext().WithCodec(cdc).WithAccountDecoder(decoder)
 		txBldr := authtxb.NewTxBuilderFromCLI()
 
-		newTx, err := utils.SignStdTx(txBldr, cliCtx, name, stdTx, viper.GetBool(flagAppend), viper.GetBool(flagOffline))
+		// if --signature-only is on, then override --append
+		generateSignatureOnly := viper.GetBool(flagRawSignature)
+		append := viper.GetBool(flagAppend) && !generateSignatureOnly
+		newTx, err := utils.SignStdTx(txBldr, cliCtx, name, stdTx, append, viper.GetBool(flagOffline))
 		if err != nil {
 			return err
 		}
+
 		var json []byte
-		if cliCtx.Indent {
+
+		switch {
```

Comment:
Nest the switch maybe? It's easier to read.

Reply:
Done, thanks